File: src/main/java/com/webversive/chroniclecompanion/enums/GameType.java
package com.webversive.chroniclecompanion.enums;

public enum GameType {
    DND("dnd"), VTM("vampire"), GEN("gen");

    public final String value;

    GameType(String value) {
        this.value = value;
    }
    public static GameType fromString(String text) {
        for (GameType b : GameType.values()) {
            if (b.value.equalsIgnoreCase(text)) {
                return b;
            }
        }
        return null;
    }

    public String getValue() {
        return this.value;
    }
}

File: src/main/app/src/data/enum/GameType.ts
enum GameType {
    VAMPIRE, DND, GEN, NULL
}
export default GameType;

export function getGameType(gameType: string): GameType {
    switch (gameType) {
        case "dnd":
            return GameType.DND;
        case "vampire":
        case "vtm":
            return GameType.VAMPIRE;
        case "gen":
            return GameType.GEN;
    }
}